pkg/backend/diy/backend.go

```diff
@@ -548,6 +548,21 @@ func massageBlobPath(path string) (string, error) {
 		path = "/" + path
 	}
 
+	// We need to set no_tmp_dir to a value to avoid using the system temp directory.
+	// See also https://github.com/pulumi/pulumi/issues/15352
+	if !strings.Contains(path, "no_tmp_dir") {
```

Comment:
We should probably url parse this, not just string.Contains otherwise you just know someone's gonna make a "file://no_tmp_dir" folder.